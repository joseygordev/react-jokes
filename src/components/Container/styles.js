export default {
  root: {
    display: 'flex',
    alignItems: 'flex-start',
    justifyContent: 'center',
    maxWidth: 980,
    width: '75%',
    height: '100vh',
    fontFamily: 'Open Sans, sans-serif',
    margin: 'auto',
  }, 
  wrapper: {
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
    justifyContent: 'space-evenly',
    width: '100%',
    paddingBottom: 32
  }
};