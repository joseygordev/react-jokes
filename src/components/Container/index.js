import React from 'react';
import PropTypes from 'prop-types';

// Styles
import styles from './styles';

const Container = ({children}) => (
  <div style={styles.root}>
    <div style={styles.wrapper}>
      {children}
    </div>
  </div>
);

Container.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired
};
Container.defaultProps = {
};

export default Container;
