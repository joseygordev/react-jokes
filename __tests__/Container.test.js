import React from 'react';
import renderer from 'react-test-renderer';
import Container from '../src/components/Container';

describe('Container', () => {
  it('Should return the Container component', () => {
    const tree = renderer
      .create(<Container>Content</Container>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
