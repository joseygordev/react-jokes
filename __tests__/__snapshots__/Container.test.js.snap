// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container Should return the Container component 1`] = `
<div
  style={
    Object {
      "alignItems": "flex-start",
      "display": "flex",
      "fontFamily": "Open Sans, sans-serif",
      "height": "100vh",
      "justifyContent": "center",
      "margin": "auto",
      "maxWidth": 980,
      "width": "75%",
    }
  }
>
  <div
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "column",
        "justifyContent": "space-evenly",
        "paddingBottom": 32,
        "width": "100%",
      }
    }
  >
    Content
  </div>
</div>
`;
